### ISSUE TITLE
`SuperExpCutoffPowerLaw4FGLSpectralModel` provides wrong results if the energy units are different from MeV

### ISSUE THREAD

**Issue by aaguasca** [gammapy/gammapy#5628]
**Gammapy version**
gammapy dev version.

**Bug description**
The class `SuperExpCutoffPowerLaw4FGLSpectralModel` provides wrong results when `__call__()` is used if the units of `energy` are different from the units of the reference energy. This affects methods like `energy_flux`, `plot`, etc.

When using `_convert_evaluate_unit()` inside `__call__()`, the units of `reference` are changed to the `energy` units. Since `expfactor` is defined in units of `reference.unit**index_2`, the units are changed, but the value of `expfactor` is not modified accordingly.  

**Expected behavior**
The value of `expfactor` should change accordingly when the units of `reference` are changed. Another option is to force it to a fixed energy unit, e.g., MeV**`index_2`.

**Possible solution**
Since accounting for varying units in `reference` would affect `_convert_evaluate_unit()` by including ad hoc code for this function, I propose to fix the units of `expfactor` to MeV**`index_2`. This approach should be the easiest solution and in line with the units provided in 4FGL DR1 and DR2.

I can take care of the PR.

**To Reproduce**
```
from gammapy.modeling.models import SuperExpCutoffPowerLaw4FGLSpectralModel
import astropy.units as u
import matplotlib.pyplot as plt

secpl=SuperExpCutoffPowerLaw4FGLSpectralModel(
    amplitude=2e-12*u.Unit("MeV-1 s-1 cm-2"),
    reference=1e+03*u.MeV,
    expfactor=5e-03,
    index_1=2e+00,
    index_2=6.6667e-01,
)

print(secpl.energy_flux(100*u.MeV,1000*u.MeV).to_value("MeV cm-2 s-1"),secpl.energy_flux(0.1*u.GeV,1*u.GeV).to_value("MeV cm-2 s-1"))

secpl.plot(sed_type="dnde",energy_bounds=[100*u.MeV,100000*u.MeV],label="OK")
secpl.plot(sed_type="dnde",energy_bounds=[100*u.MeV,100*u.GeV],label="wrong")
plt.legend()
```

**Comment by registerrier** [comment]
Is there a reason why you have to change the `reference` unit? A priori the Fermi catalog units are frozen.

**Comment by aaguasca** [comment]
Sorry, I was not clear enough. The problem is related to having `expfactor` in units of `reference`**`index_2` and `__call__()` changing the units of `reference`.

Let me give you an example using the code to reproduce this bug, when you use a method that involves `__call__()`, e.g., `secpl.energy_flux(0.1*u.GeV,1*u.GeV)`,  the unit of `reference` (MeV) is modified to the unit of `energy` (GeV for this particular case) 

https://github.com/aaguasca/gammapy/blob/22632263f99ef0a547de4480f6ea489795b95c11/gammapy/modeling/models/spectral.py#L144

As a consequence, when using `evaluate`, the function uses the value of `expfactor` with the new unit of `reference` (5e-03 **GeV****`index_2` in this case). This value is different from the original one, 5e-03 **MeV****`index_2`. Therefore providing wrong results.

This issue does not happen if `energy` has the same unit as `reference`, but since the user defines the units of `energy`, this error will occur even if the Fermi catalog units are frozen.

**Comment by registerrier** [comment]
OK thanks. Then I suppose that one has to overload `SpectralModel._convert_evaluate_energy()` for this specific class to make sure it converts to MeV, no?

**Comment by aaguasca** [comment]
Yes, one would have to store the original unit of `reference` to convert `expfactor` to the new ones. This will overload `SpectralModel._convert_evaluate_unit()` for this specific class.

An alternative solution is to hardcode the units of `expfactor` to MeV**`index_2`. This agrees with the units used in 4FGL DR1 and DR2, which are the catalogues that rely on this function. If the docs are modified, I think it would not be an issue for users interested in using this function for other purposes.

I prefer the alternative solution. I can take care of the PR.
